### PR TITLE
[stable9] Birthday calendar should never have write acl 

### LINKED ISSUE #24154: Can create events in contacts_birthday calendar
### Steps to reproduce
1. Use calendar app
2. Add a new event

### Expected behaviour
Calendar should not be selectable at all

### Actual behaviour
200

Also happens on 9.0

### PR DESCRIPTION
**Pull request by DeepDiver1975**
fixes #24154
backport of https://github.com/owncloud/core/pull/24155

### CODE REVIEW

File: apps/dav/tests/unit/caldav/caldavbackendtest.php [hunk 1]
```diff
@@ -25,6 +25,7 @@
 use OCA\DAV\CalDAV\CalDavBackend;
 use OCA\DAV\CalDAV\Calendar;
 use OCA\DAV\Connector\Sabre\Principal;
+use OCP\IL10N;
```
Comment (nickvergessen) [inline apps/dav/tests/unit/caldav/caldavbackendtest.php]: :speak_no_evil:

**Comment by nickvergessen** [comment]
:+1:

**Comment by nickvergessen** [comment]
Summoning @schiesbn for the backport

**Comment by schiesbn** [comment]
:+1:

**Pull request merged**